rfc/rfc021-vp_token-grant-type.md

```diff
@@ -0,0 +1,189 @@
+5. The `kid` field MUST be a DID URL and MUST resolve to a verificationMethod in the DID Document. The DID part of the DID URL MUST match the `iss` field.
+6. The `sub` field MUST match the `credentialSubject.id` field from all the Verifiable Credentials that are used to request the access token.
+7. The `aud` field MUST match the DID of the Authorization Server.
+
+### 4.3 JSON format requirements
+
+1. The assertion MUST be a valid JSON object according to §4.10 the Verifiable Credentials Data Model 1.1 [VC-DATA-MODEL].
+2. The proof of the VP MUST be a valid [JSONWebSignature2020] object.
+3. The `challenge` field of the JSON object MUST be a string that is unique for each token request.
+4. The `domain` field of the JSON object MUST be a DID under control of the Authorization Server.
+5. The `verificationMethod` field of the Proof MUST be a DID URL.
+6. The `holder` field if present MUST match the `credentialSubject.id` field from all the Verifiable Credentials that are used to request the access token.
+7. The `verificationMethod` field of the proof MUST match the `credentialSubject.id` field from all the Verifiable Credentials that are used to request the access token.
+
+## 5. Presentation Definition endpoint
+
+In order for a client to know which Presentation Definition [PE] to use, the Authorization Server MUST provide a Presentation Definition endpoint.
+The Presentation Definition endpoint MUST be registered as a `presentation_definition_endpoint` in the Authorization Server metadata [RFC8414].
+The Presentation Definition endpoint MUST return a Presentation Definition that corresponds with the requested scope.
+The endpoint has a single query parameter `scope` that contains the requested scope. The parameter may contain multiple values.
+
+The following example shows a request to the Presentation Definition endpoint:
+
+    GET /presentation_definition?scope=a&scope=b HTTP/1.1
+    Host: as.example.com
+
+## 6. Access Token Introspection
+
+The Authorization Server MAY support an access token introspection endpoint as defined in OAuth 2.0 [RFC7662].
+The introspection endpoint SHOULD map the fields as follows:
+
+* `iss`: The issuer of the access token (DID).
+* `sub`: The subject of the access token (DID).
+* `exp`: The expiration time of the access token.
+* `iat`: The time the access token was issued.
+* `scope`: The granted scope.
+* `vcs`: The Verifiable Credentials that were used to request the access token using the same encoding as used in the access token request.
+
+## 7. Security Considerations
+
+The nonce/challenge (nonce) is used to prevent replay attacks. The nonce is a random string that is generated by the client.
+The nonce is included in the signed data. The Authorization Server MUST reject any request that uses a nonce that has been used before.
+The Authorization Server MAY reduce storage requirements by only storing the nonce for the lifetime of the Verifiable Presentation.
```

Comment:
That's not possible within this flow. It could be possible if the PD endpoint returns the nonce.
The spec should then add how long that nonce can be valid for.